pkg/api/dryrunclient.go

```diff
@@ -162,7 +169,14 @@ func (d *DryRunClient) ImageBuild(ctx context.Context, reader io.Reader, options
 }
 
 func (d *DryRunClient) ImageInspectWithRaw(ctx context.Context, imageName string) (moby.ImageInspect, []byte, error) {
-	return moby.ImageInspect{ID: "dryRunId"}, nil, nil
+	caller := getCallingFunction()
+	switch caller {
+	case "pullServiceImage", "buildContainerVolumes":
```

Comment:
It's not enough for example if you run the `create` command, the `ImageInspectWithRaw` function will be call multiple times, first to check if the image is present locally, so we want to get an error if the image doesn't exists to trigger a pull and after the pull (and this time we don't want to fail because we just faked the pull)

Reply:
nice... which seems to demonstrate some design issue to be fixed, as we should not require a second call after pull :P
anyway, go for it